segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/TableCompactor.java

```diff
@@ -129,7 +129,7 @@ protected CompactionArgs newCompactionArgs(long startOffset) {
     CompletableFuture<Boolean> isCompactionRequired() {
         final long startOffset = getCompactionStartOffset();
         final long lastIndexOffset = getLastIndexedOffset();
-        if (startOffset + this.config.getMaxCompactionSize() >= lastIndexOffset) {
+        if (!this.config.isCompactionEnabled() || startOffset + this.config.getMaxCompactionSize() >= lastIndexOffset) {
```

Comment:
I think it's better if we have a quick return at the very beginning of the `isCompactionRequired`. The logic would be like this: 
```
  if (!this.config.isCompactionEnabled()) {
           return CompletableFuture.completedFuture(false);
  }

//// Compaction is enabled, so the next step is to do what you want to do.

        final long startOffset = getCompactionStartOffset();
        final long lastIndexOffset = getLastIndexedOffset();
...............
```

It's almost the same as your current change, just a little improvement, but I think it could make code clear to read.